frontend/apps/reader/modules/readerview.lua

```diff
@@ -792,7 +792,19 @@ function ReaderView:onReadSettings(config)
     self.highlight.saved = config:readSetting("highlight", {})
     -- Highlight formats in crengine and mupdf are incompatible.
     -- Backup highlights when the document is opened with incompatible engine.
-    local _, page_highlights = next(self.highlight.saved) -- get the first page with highlights
+    local page, page_highlights
+    while true do -- remove empty tables for pages without highlights and get the first page with highlights
+        page, page_highlights = next(self.highlight.saved)
+        if page then
+            if #page_highlights == 0 then
+                self.highlight.saved[page] = nil
+            else
+                break
+            end
+        else
+            break
+        end
```

Comment:
I'm all for readability and simplicity, and you're all for short stuff - but this `else break else break` feels a bit to long and noisy - for such a minor bit of code.
May we can turn it around, if you feel it reads ok (I do):
```lua
if not page or #page_highlights > 0 then
    break -- we're done (there is none, or there is some usable)
else
    self.highlight.saved[page] = nil -- clean it up while we're at it, and find another one
end
```